tests/gis_tests/rasterapp/test_rasterfield.py

```diff
@@ -0,0 +1,100 @@
+import json
+
+from django.contrib.gis.db import models
+from django.contrib.gis.shortcuts import numpy
+from django.core.exceptions import ImproperlyConfigured
+from django.test import (
+    TestCase, TransactionTestCase, skipIfDBFeature, skipUnlessDBFeature,
+)
+
+from ..data.rasters.textrasters import JSON_RASTER
+from .models import RasterModel
+
+
+@skipUnlessDBFeature('supports_raster')
+class RasterFieldTest(TransactionTestCase):
+    available_apps = ['gis_tests.rasterapp']
+
+    def test_field_null_value(self):
+        "Testing creating model where the RasterField has value null."
+        # Create model instance with Null value
+        rastermodel = RasterModel.objects.create(rast=None)
+        rastermodel = RasterModel.objects.get(id=rastermodel.id)
+        self.assertIsNone(rastermodel.rast)
+
+    def test_model_creation(self):
+        "Testing RasterField through a test model."
+        # Create model instance from JSON raster
+        rastermodel = RasterModel.objects.create(rast=JSON_RASTER)
+        rastermodel = RasterModel.objects.get(id=rastermodel.id)
+
+        # Test raster metadata properties
+        self.assertEqual((5, 5), (rastermodel.rast.width, rastermodel.rast.height))
+        self.assertEqual([0.0, -1.0, 0.0, 0.0, 0.0, 1.0],
+                         rastermodel.rast.geotransform)
+
+        self.assertEqual(None, rastermodel.rast.bands[0].nodata_value)
+
+        # Compare srs
+        self.assertEqual(rastermodel.rast.srs.srid, 4326)
+
+        # Compare pixel values
+        band = rastermodel.rast.bands[0].data()
+
+        # If numpy, convert result to list
+        if numpy:
+            band = band.tolist()
+
+        # Loop through rows in band data and assert single
+        # value is as expected.
+        self.assertEqual(
+            [[0.0, 1.0, 2.0, 3.0, 4.0],
+             [5.0, 6.0, 7.0, 8.0, 9.0],
+             [10.0, 11.0, 12.0, 13.0, 14.0],
+             [15.0, 16.0, 17.0, 18.0, 19.0],
+             [20.0, 21.0, 22.0, 23.0, 24.0]],
+            band
```

Comment:
Thanks for pointing this out. This is expected behaviour, I think raster data is usually stored as straight arrays, so the flat array is the "original" format. That is why in my opinion the list should be flat and leave it to the user to convert into a nested list if that is desired. In numpy, specifying the shape is not a costly operation (correct me if I am wrong here), so the shape can be specified for convenience, as it prints nicely and its easier to access specific pixel values.

So I added the `flatten()` argument to the numpy case for the test to work without numpy as well.